velero-plugins/imagecopy/imagestream.go

```diff
@@ -12,10 +14,47 @@ import (
 	"github.com/containers/image/v5/signature"
 	"github.com/containers/image/v5/transports/alltransports"
 	"github.com/go-logr/logr"
+	"github.com/kaovilai/udistribution/pkg/image/udistribution"
 	imagev1API "github.com/openshift/api/image/v1"
 	//"github.com/sirupsen/logrus"
 )
 
+const (
+	EnvOpenShiftImagestreamBackup = "OPENSHIFT_IMAGESTREAM_BACKUP"
+	// Prefix to indicate use of plugin registry
+	BSLRoutePrefix				  = "bsl://"
+)
+
+// We expect usePluginRegistry ENV to be set once when container is started.
+// When true, we will use the plugin registry to copy images.
+var usePluginRegistry, _ =  strconv.ParseBool(os.Getenv(EnvOpenShiftImagestreamBackup))
+
+// use getter to avoid changing bool in other packages
+func UsePluginRegistry() bool {
+	return usePluginRegistry
+}
+
+// Using struct for options clarity when specifying options
+type CopyLocalImageStreamImagesOptions struct {
+	InternalRegistryPath string
+	SrcRegistry string
+	DestRegistry string
+	DestNamespace string
+	CopyOptions *copy.Options
+	Log logr.Logger
+	UpdateDigest bool
+	Ut *udistribution.UdistributionTransport
+}
+
+func (o CopyLocalImageStreamImagesOptions) GetSrcRegistry() string {
+	return o.SrcRegistry
+}
+
+func (o CopyLocalImageStreamImagesOptions) GetDestRegistry() string {
+	return o.DestRegistry
+}
+
+
 // CopyLocalImageStreamImages copies all local images associated with the ImageStream
 // is: ImageStream resource that images are being copied for
 // internalRegistryPath: The internal registry path for the cluster in which is comes from, used to determine which images are local
```

Comment:
NIT: Can you just add a small comment that the params are grouped under a struct called `CopyLocalImageStreamImagesOptions`

Reply:
+1